### ISSUE TITLE
issue about the tokenpruner

### ISSUE THREAD

**Issue by JuilieZ** [LiAutoAD/LightVLA#15]
Hi, thank you for your wonderful work on "LightVLA" — it's been very helpful for us.

Recently, however, I encountered some confusion regarding the **TokenPruner** module during training. Specifically, in the following code snippet:
`indices, patches = self.score_to_indices(score, patches)
batch_indices = torch.arange(patches.shape[0], device=patches.device).unsqueeze(1).expand_as(indices)

tokens = torch.cat([cls_token, patches, task], dim=1)

patches_id = patches_id[batch_indices, indices].view(bsz, -1)
position_ids = torch.cat([cls_token_id, patches_id, task_id], dim=1)

if attention_mask is not None:
    patches_mask = patches_mask[batch_indices, indices].view(bsz, -1)
    attention_mask = torch.cat([cls_token_mask, patches_mask, task_mask], dim=1)

return tokens, position_ids, attention_mask`

Since `patches_mask` is typically an all-`True` tensor before pruning, the line  
`patches_mask = patches_mask[batch_indices, indices].view(bsz, -1)`  
would still result in an all-`True` tensor.  

I’m concerned that this might cause some tokens to be computed repeatedly in the subsequent transformer layers. Could this lead to incorrect computation?
